crates/bitwarden-crypto/src/store/context.rs

```
@@ -149,6 +164,30 @@ impl<Ids: KeyIds> KeyStoreContext<'_, Ids> {
         self.local_asymmetric_keys.retain(f);
     }
 
+    fn drop_symmetric_key(&mut self, key_id: Ids::Symmetric) -> Result<()> {
```

These as currently written will silently fail to delete the key if passed a non-local key while using a read-only context. We should probably change the implementation to be more like:

```rust
    fn drop_symmetric_key(&mut self, key_id: Ids::Symmetric) -> Result<()> {
        if key_id.is_local() {
            self.local_symmetric_keys.remove(key_id);
        } else {
            self.global_keys.get_mut()?.symmetric_keys.remove(key_id);
        }
        Ok(())
    }
```